aggregator/src/aggregator.rs

```diff
@@ -1179,19 +1226,52 @@ impl VdafOps {
                 }
             };
 
-        let encoded_leader_plaintext_input_share = match hpke::open(
-            hpke_keypair.config(),
-            hpke_keypair.private_key(),
-            &HpkeApplicationInfo::new(&Label::InputShare, &Role::Client, task.role()),
-            leader_encrypted_input_share,
-            &InputShareAad::new(
-                *task.id(),
-                report.metadata().clone(),
-                report.public_share().to_vec(),
-            )
-            .get_encoded(),
-        ) {
-            Ok(encoded_leader_plaintext_input_share) => encoded_leader_plaintext_input_share,
+        let try_hpke_open = |hpke_keypair: &HpkeKeypair| -> Result<Vec<u8>, Error> {
+            match hpke::open(
+                hpke_keypair.config(),
+                hpke_keypair.private_key(),
+                &HpkeApplicationInfo::new(&Label::InputShare, &Role::Client, task.role()),
+                leader_encrypted_input_share,
+                &InputShareAad::new(
+                    *task.id(),
+                    report.metadata().clone(),
+                    report.public_share().to_vec(),
+                )
+                .get_encoded(),
+            ) {
+                Ok(results) => Ok(results),
+                Err(error) => Err(error.into()),
+            }
```

Comment:
I think you can do `Result::map_err` instead of writing out this `match`.

Reply:
I'm surprised this isn't a clippy lint!